### ISSUE TITLE
Wallet does not recognize its own change address

### ISSUE THREAD

**Issue by pinheadmz** [handshake-org/hsd#411]
There have been at least two separate reports of this bug, and I've heard that it may have happened to the Purse bcoin wallet at least once in the past. It's very difficult to diagnose or reproduce but what seems to happen is this:

- The wallet sends a transaction and sends the change output to the next address in the BIP44 `change` path.
- When this transaction is added to the wallet, the change address is not recognized as being owned by the wallet, and has no path information.
- The wallet may continue to use the "ghost" address for change and does not increment the change address index of the wallet account

Additional related behavior: the wallet does not know its own private key for the address:
```
$ hsw-cli change --account=default
{... "address": address...}

$ hsw-cli key <address>
null
```

Even though the wallet may provide the change address again 

**Workaround:** Your coins are not lost, but so far the only way we have been able to recover a wallet in this state is to import the seed into a new wallet and rescan:

```
$ hsw-cli mkwallet --id=recovery --mnemonic="abandon abandon abandon ..."
$ hsw-cli --id=recovery rescan
```

**Comment by pinheadmz** [comment]
This issue may persist on a wallet restored by mnemonic. This user had large gaps in their receive addresses and so I instructed them to manually regenerate addresses until the appropriate index had been reached and rescan again:

> I recovered the wallet, and use generate receiving and change address, and they are same with the original wallet, which contains only 54 received addresses and 26 changed addresses, and the 15, 18, 26 index’s key is lost

**Comment by pinheadmz** [comment]
Closed by #414
